src/stateful-editor-page.jsx

```diff
@@ -18,18 +19,31 @@ var StatefulEditorPage = React.createClass({
 
     getDefaultProps: function() {
         return {
-            componentClass: EditorPage
+            componentClass: EditorPage,
+            searchString: "",
+            searchIndex: 0
         };
     },
 
     render: function() {
-        return <this.props.componentClass {...this.state} />;
+        return <div>
+            <this.props.componentClass {...this.state} />
+            <SearchAndReplaceDialog
+                ref="searchAndReplace"
+                question={this.state.question}
+                hints={this.state.hints}
+                particle={this.state.json}
+                searchString={this.state.searchString}
+                searchIndex={this.state.searchIndex}
+                onChange={this.handleSearchChange} />
```

Comment:
This should go in EditorPage or ArticleEditor. StatefulEditorPage is just a wrapper around those two prop-based components for rendering as a top level component, like test.html or article.html does, and isn't actually used on khanacademy.org.

It's still find to have state on EditorPage and ArticleEditor, as long as we don't expect that state to be persisted.